Documentation/troubleshooting.md

```diff
@@ -1,4 +1,84 @@
-### Firewalls
+# Troubleshooting
+
+# General
+
+## Logging
+Flannel uses the `glog` library but only supports logging to stderr. The severity level can't be changed but the verbosity can be changed with the `-v` option. flannel does not make extensive use of the verbosity level but increasing th value from `0` (the default) will result in some additional logs. To get the most detailed logs, use `-v=10`
+
+```
+-v value
+    	log level for V logs
+-vmodule value
+    	comma-separated list of pattern=N settings for file-filtered logging
+-log_backtrace_at value
+    	when logging hits line file:N, emit a stack trace
+```
+
+When running under systemd (e.g. on CoreOS container linux) the logs can be viewed with `journalctl -u flanneld`
+
+When flannel is running as a pod on Kubernetes, the logs can be viewed with `kubectl logs --namespace kube-system <POD_ID> -c kube-flannel`. You can find the pod IDs with `kubectl get po --namespace kube-system -l app=flannel`
+
+## Interface selection and the public IP.
+Most backends require that each node has a unique "public IP" address. This address is chosen when flannel starts, and if the address changes then flannel needs to be restarted. Since leases are tied to the public address
```

Comment:
This address is chosen when flannel starts. Because leases are tied to the public address, if the address changes, flannel must be restarted.